src/target/llvm/llvm_module.cc

```diff
@@ -485,6 +485,11 @@ TVM_REGISTER_GLOBAL("target.llvm_lookup_intrinsic_id")
       return static_cast<int64_t>(llvm::Function::lookupIntrinsicID(name));
     });
 
+TVM_REGISTER_GLOBAL("target.llvm_get_intrinsic_name")
+    .set_body_typed([](int64_t id) -> String {
+      return std::string(llvm::Intrinsic::getName(static_cast<llvm::Intrinsic::ID>(id)));
```

Comment:
The function `getName` is overloaded, and the version that only takes the intrinsic id will not work with overloaded LLVM intrinsics (such as `llvm.ctpop.i32`).  With the newer LLVM it's better to use `getBaseName`, but that function was only added in LLVM13.  In the older LLVMs, this function has changed several times, so you may need to add specific #if's for different versions.  You can check for overloading with `isOverloaded`.  To get the exact overloaded name, you'd need to generate the LLVM types for the arguments, so that may be too involved for just printing the TIR.

Reply:
I've switched to using `getBaseName` when llvm >= 13. On lower versions of llvm I still use `getName`, but I use the overloaded types version. When provided with an empty array of types, this version will return the generic name which I think is what we want.